mlflow/gateway/client.py

```diff
@@ -110,6 +110,18 @@ def search_routes(self, page_token: Optional[str] = None) -> PagedList[Route]:
         response_json = self._call_endpoint(
             "GET", MLFLOW_GATEWAY_CRUD_ROUTE_BASE, json_body=json.dumps(request_parameters)
         ).json()
+
+        # NB: avoiding a comprehension so as not to double the memory allocation
+        for resp in response_json["routes"]:
+            # Assemble the URI path from the base route and the route name
+            uri_path = assemble_uri_path([MLFLOW_GATEWAY_CRUD_ROUTE_BASE, resp["name"]])
+
+            # Assemble the full URL from the gateway URI and the URI path
+            full_url = join_url(self._gateway_uri, uri_path)
```

Comment:
@BenWilson2 we should be returning the route URL from the backend (FastAPI) server as part of the GetRoute / SearchRoutes response. For the FastAPI server, we should just return the path suffix. In the client, we can check to see if the URL is a suffix or a full URL. If it's a suffix, we can append it to the gateway URI using `append_to_uri_path()`.

We shouldn't have the client hardcode the route URL definition, which may differ from the route URL definition on Databricks.

In particular, this code on line 120 won't work for Databricks, right? Since `self._gateway_uri` is the literal string `databricks`.